src/main/java/com/nasrpi/home/HomeController.java

```diff
@@ -50,14 +63,26 @@ public boolean createDirectory(@RequestBody final String directoryPath) {
 		return homeRepository.createDirectory(directoryPath);
 	}
 
-	@RequestMapping(value = "/moveFolder", method = RequestMethod.POST, produces = "application/json")
+	@RequestMapping(value = "/moveFolder", method = RequestMethod.POST, produces = KeyConstants.APPLICATION_JSON)
 	public boolean moveFolder(@RequestBody final MoveModel moveModel) {
 		return homeRepository.moveFolder(moveModel.getSource(), moveModel.getDestination());
 	}
-	
-	@RequestMapping(value = "/moveFile", method = RequestMethod.POST, produces = "application/json")
+
+	@RequestMapping(value = "/moveFile", method = RequestMethod.POST, produces = KeyConstants.APPLICATION_JSON)
 	public boolean moveFile(@RequestBody final MoveModel moveModel) {
 		return homeRepository.moveFile(moveModel.getSource(), moveModel.getDestination());
 	}
 
+	@PostMapping("/uploadFile")
+	public UploadFileResponse uploadFile(@RequestParam("file") MultipartFile file,
+			@RequestParam("path") final String uploadPath) {
+		return homeRepository.uploadFile(file, uploadPath, fileStorageService);
+	}
+
+	@GetMapping("/downloadFile")
+	public ResponseEntity<Resource> downloadFile(@RequestParam("path") final String filePath,
```

Comment:
can we use @RequestBody and create a model for all the params passed instead of @RequestParam ?
generally its difficult (not impossible) to send request params from ui

Reply:
done. changed the GetMapping to PostMapping  as a get request cannot have a body.